src/region/RegionAnnotations.tsx

```diff
@@ -1,83 +1,130 @@
 import * as React from 'react';
-import { Annotation, Rect, TargetRegion } from '../@types';
+import merge from 'lodash/merge';
 import PopupReply from '../components/Popups/PopupReply';
 import RegionAnnotation from './RegionAnnotation';
 import RegionCreator from './RegionCreator';
+import { Annotation, Rect, NewAnnotation, TargetRegion } from '../@types';
+import { CreatorItem } from '../store/creator';
 import './RegionAnnotations.scss';
 
 type Props = {
     annotations: Annotation[];
     isCreating: boolean;
+    page: number;
+    scale: number;
+    saveAnnotation: (annotation: NewAnnotation) => void;
+    setStaged: (staged: CreatorItem | null) => void;
+    setStatus: (status: string) => void;
+    staged?: CreatorItem | null;
+    status: string | null;
 };
 
 type State = {
-    current: Partial<Rect> | null;
-    hasComment: boolean;
-    hasDrawn: boolean;
+    targetRef?: HTMLAnchorElement;
 };
 
 export default class RegionAnnotations extends React.Component<Props, State> {
     static defaultProps = {
         annotations: [],
         isCreating: false,
+        scale: 1,
     };
 
-    targetRef: React.RefObject<HTMLAnchorElement> = React.createRef();
+    state: State = {};
 
-    state: State = {
-        current: null, // TODO: Move unsaved region state to central store
-        hasComment: false,
-        hasDrawn: false,
-    };
+    setStaged(data: Partial<CreatorItem>): void {
+        const { staged, page, setStaged } = this.props;
+        setStaged(merge({ location: page }, staged, data));
+    }
+
+    setStatus(status: string): void {
+        const { setStatus } = this.props;
+        setStatus(status);
+    }
 
     handleCancel = (): void => {
         this.reset();
     };
 
     handleChange = (text?: string): void => {
-        this.setState({
-            hasComment: !!text,
-        });
+        this.setStaged({ message: text });
     };
 
     handleSubmit = (): void => {
-        // TODO: Update store with new shape/text
+        const { staged, saveAnnotation } = this.props;
+
+        if (!staged) {
+            return;
+        }
+
+        const { location, message = '', shape } = staged;
+
+        saveAnnotation({
```

Comment:
Is this the same annotation shape as the API being constructed here? Should that be done elsewhere like the action or some utility?

Reply:
I mean, I can pass each value to the action and have it construct the payload, but we'd have to pass the following information for each save:

- message
- location type
- location value
- annotation type (region, highlight, etc.)
- annotation shape(s)

I'm not sure how much we save by moving the logic elsewhere.

Reply:
Maybe move it to a util method then? I imagine we'll need it elsewhere in the future as well

Reply:
Moved to a region-specific async action. Thanks for the suggestion!